cmd/dfdaemon/app/root.go

```diff
@@ -0,0 +1,107 @@
+package app
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/dragonflyoss/Dragonfly/cmd/dfdaemon/options"
+	"github.com/dragonflyoss/Dragonfly/dfdaemon/initializer"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	opt           = &options.Options{}
+	defaultDFRepo = filepath.Join(os.Getenv("HOME"), ".small-dragonfly/dfdaemon/data/")
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "dfdaemon",
+	Short: "The dfdaemon is a proxy that intercepts image download requests.",
+	Long:  "The dfdaemon is a proxy between container engine and registry used for pulling images.",
+	Run: func(cmd *cobra.Command, args []string) {
+		runDaemon()
+	},
+}
+
+func init() {
```

Comment:
Nit: I do not see any problem if we put `setupFlags` right before `runDaemon` at L28, or is there any? (If we do not have to use `init`, we don't).

Reply:
Can you explain the reason why we should avoid using `init`?

Reply:
Maybe it's not bad to use `init` according to [cobra example](https://github.com/spf13/cobra/blob/master/cobra/cmd/root.go#L42).

Reply:
Well, I am not saying we should avoid using `init`, but should avoid abusing it. Basically, `init` is used for side effects(plugin registration, global initialization, etc). Another argument could be: What if we do not want to run this piece of code in unit tests(I guess we certainly do not want to run this code in uni tests)?

Reply:
I think both are OK. While the way @zhouhaibing089 suggested is much straight-forward.